text/0000-pre-elves_soft.md

```diff
@@ -0,0 +1,116 @@
+
+## Performance, Ergonomics, and Compatibility
+
+We expect early soft concensus introduce back pressure that radically alters performance.  We no longer run approvals checks upon all forks.  As primary slots occur once every other slot in expectation, one might expect a 25% reduction in CPU load, but this depends upon diverse factors.
+
+We apply back pressure by dropping some whole relay chain blocks though, so this shall increase the expected parachain blocktime somewhat, but how much depens upon future optimisation work.
+
+### Compatibility
+
+Major upgrade
+
+## Prior Art and References
+
+...
+
+## Unresolved Questions
+
+Provide specific questions to discuss and address before the RFC is voted on by the Fellowship. This should include, for example, alternatives to aspects of the proposed design where the appropriate trade-off to make is unclear.
+
+## Future Directions and Related Material
+
+### Sassafras
+
+Arguably, a sassafras RC like JAM could avoid `preferred_fork` flag, by only releasing availability votes for at most one sassafras equivocation.  We wanted availability for babe forks, but sassafras has only equivocations, so those block can simply be dropped.
+
+In principle, a sassafras equivocation could still enter the valid chain, assuming 2/3rd of validators provide availability votes for the same equivocations. If JAM lacks the `preferred_fork` flag then enactment proceeds slower in this case, but this should almost never occur.
+
+### Thresahold randomness
+
+We think threshold randomness could reduce the tranche zero approcha checker assigments by roughly 40%, meaning a fixed 15 vs the expected 25 in the elves paper (30 in production now).
+
+We do know threshold VRF based schemes that address relay chain equivocations directly, by using as input the relay chain block hash.  We have many more options with early soft concensus though.  TODO  In particular, we only know two post-quantum approaches to elves, and the bandwidth efficent one needs early soft concensus.
+
+### Mid-strenght concensus
+
+In this RFC, we only require that each relay chain block contain preference votes for its parent from 2/3rds of validators.  We could enforce the opposite direction too:  Around y>2 seconds after a validator V has seen preference votes for a chain head X from 2/3rd of validators, the V begins rejecting any relay chain block that does not build upon X.  This is tricky because the y>2 second delay must be long enough so that most honest nodes learn both X and its preference votes.  This strengthens MEV defenses that assume some honest nodes.
+
```

Comment:
Indeed we can use this vote in an economic finality mechanism that gives new trade-offs of latency for security. In 8-10 seconds after a block is produced, we would normally know that only disputes, which always result in large slashes could reverse it. There should be an easy migration path from this RFC to such a protocol, by putting more conditions on this vote and block production.